lib/carma/client/mule_soft_auth_token_client.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'carma/client/mule_soft_auth_token_configuration'
+
+module CARMA
+  module Client
+    class MuleSoftAuthTokenClient < Common::Client::Base
+      include Common::Client::Concerns::Monitoring
+
+      STATSD_KEY_PREFIX = 'api.carma.mulesoft.auth'
+
+      configuration MuleSoftAuthTokenConfiguration
+
+      class GetAuthTokenError < StandardError; end
+
+      def new_bearer_token
+        with_monitoring do
+          response = perform(:post,
+                             'oauth2/ause1x1h6Zit9ziQL0j6/v1/token',
```

Comment:
🤔 That endpoint looks oddly like a key in its own right - the part between oauth2/ and /v1/token isn't sensitive, right?

Reply:
You need to have the client_id and client_secret to get a valid token back, and I was told it was the same url for all environments. I'll double check if it is sensitive or not, but I'm not totally sure what that would accomplish. I agree having an id felt weird, but I think it may just be unique for different tools that they are using to authorize with it?

Reply:
huh. Surprised by that design decision vs using a readable URL, but presumably there's a good reason behind it.